src/main/java/org/apache/commons/imaging/formats/gif/GifImageParser.java

```diff
@@ -492,17 +502,29 @@ public Dimension getImageSize(final ByteSource byteSource, final Map<String, Obj
             throw new ImageReadException("GIF: Couldn't read ImageDescriptor");
         }
 
-        // Prefer the size information in the ImageDescriptor; it is more
-        // reliable
-        // than the size information in the header.
```

Comment:
We will need to understand why the old code was using the other width/height too.

Reply:
Thanks for the description and link! Will read it when reviewing it again :tada:

Reply:
Added a comment in code and a unit test